clients/src/main/java/org/apache/kafka/common/network/KafkaChannel.java

```diff
@@ -236,6 +241,27 @@ public ChannelMuteState muteState() {
         return muteState;
     }
 
+    /**
+     * Mute this channel on {@link #prepare()} failure. This will remove all operations from this channel.
+     */
+    private void muteOnPrepareFailure() {
+        mute();
+        transportLayer.removeInterestOps(SelectionKey.OP_WRITE | SelectionKey.OP_CONNECT | SelectionKey.OP_ACCEPT);
```

Comment:
`Mute` is misleading since we are actually delaying writes. Perhaps rename the method to indicate that? And we don't need to remove OP_CONNECT/OP_ACCEPT since they wont be set when we get here.